sql/core/src/main/scala/org/apache/spark/sql/execution/arrow/ArrowConverters.scala

```diff
@@ -110,6 +110,8 @@ private[sql] object ArrowConverters extends Logging {
       val writeChannel = new WriteChannel(Channels.newChannel(out))
 
       Utils.tryWithSafeFinally {
+        arrowWriter = ArrowWriter.create(root)
```

Comment:
so we always create a new writer when returning an iterator. is it because the writer is not re-usable?

Reply:
They "can" be reused. This change intentionally doesn't reuse it to release the buffers early.

Reply:
then why do we need to create new writers if buffers are already released by `root.clear()`?